tests/smoke-tests/test-randomness-consistency.ts

```diff
@@ -0,0 +1,173 @@
+        args: [],
+        pageSize: limit,
+        startKey: last_key,
+      });
+
+      if (query.length == 0) {
+        break;
+      }
+      count += query.length;
+
+      // TEMPLATE: convert the data into the format you want (usually a dictionary per account)
+      for (const request of query) {
+        const key = request[0].toHex();
+        expect(key.length >= 18, "storage key should be at least 64 bits"); // assumes "0x"
+
+        const requestIdEncoded = key.slice(-16);
+        const requestId = hexToBigInt(requestIdEncoded, { isLe: true });
+
+        requests.push({ id: Number(requestId), state: request[1] });
+        numRequests += 1;
+        last_key = key;
+      }
+
+      // Debug logs to make sure it keeps progressing
+      // TEMPLATE: Adapt log line
+      if (true || count % (10 * limit) == 0) {
+        debug(`Retrieved ${count} requests`);
+        debug(`Requests: ${requests}`);
+      }
+    }
+
+    requestCount = ((await apiAt.query.randomness.requestCount()) as any).toNumber();
+
+    // TEMPLATE: Adapt proxies
+    debug(`Retrieved ${count} total proxies`);
+  });
+
+  it("should have fewer Requests than RequestCount", async function () {
+    this.timeout(10000);
+
+    const numOutstandingRequests = numRequests;
+    expect(numOutstandingRequests).to.be.lessThanOrEqual(requestCount);
+  });
+
+  it("should not have requestId above RequestCount", async function () {
+    this.timeout(1000);
+
+    const highestId = requests.reduce((prev, request) => Math.max(request.id, prev), 0);
+    expect(highestId).to.be.lessThanOrEqual(requestCount);
+  });
+
+  it("should not have results without a matching request", async function () {
+    this.timeout(10000);
+
+    let query = await apiAt.query.randomness.randomnessResults.entries();
+    await query.forEach(([key, results]) => {
+      // offset is:
+      // * 2 for "0x"
+      // * 32 for module
+      // * 32 for method
+      // * 16 for the hashed part of the key: the twox64(someRequestType) part
+      // the remaining substr after offset is the concat part, which we can decode with createType
+      const offset = 2 + 32 + 32 + 16;
+      const requestTypeEncoded = key.toHex().slice(offset);
+      const requestType = context.polkadotApi.registry.createType(
+        `PalletRandomnessRequestType`,
+        "0x" + requestTypeEncoded
+      );
+
+      // sanity check
+      expect(
+        (requestType as any).isBabeEpoch || (requestType as any).isLocal,
+        "unexpected enum in encoded RequestType string"
+      );
+
+      if ((requestType as any).isBabeEpoch) {
+        let epoch = (requestType as any).asBabeEpoch;
+        // TODO
+      } else {
+        // look for any requests which depend on the "local" block
+        let block = (requestType as any).asLocal;
+        let found = requests.find((request) => {
+          // TODO: can we traverse this hierarchy of types without creating each?
```

Comment:
Could use review

Reply:
We could but it would require keeping the polkadotjs types.
Also the impact is very low to keep it in the loop I believe so it is easier to read that way.
I'll probably copy a part of it in the template to show others how you did it